src/session_impl.cpp

```diff
@@ -3145,7 +3133,8 @@ namespace aux {
 			for (auto& ext : m_ses_extensions)
```

Comment:
in this case it most likely won't make a difference, but imagine you iterate over a large list of items and check if some bool is true on each of them. For the ones where it's true, you perform some work. You're going to pay the cost of pulling in a cache line for each test, making it scale O(n). If the objects instead were sorted or organized based on the property or type (and perhaps even organized contiguously in memory) it would be a lot cheaper to iterate over them.

in short, the last-minute-check indicates that data/objects may not be organized optimally in memory.